roles/postgres/tasks/setup_db.yml

```diff
@@ -0,0 +1,63 @@
+---
+- name: Include OS specific variables
+  ansible.builtin.include_vars: "{{ item }}"
+  loop: >-
+    {{ lookup('first_found', os_fallback, errors='ignore', wantlist=True) }}
+
+- name: Configure postgresql client auth
+  become: true
+  notify:
+    - Restart-postgresql
+  ansible.builtin.blockinfile:
+    path: "{{ postgresql_conf_path }}/pg_hba.conf"
+    block: |
+      {% for host in postgres_db_clients | map('extract', hostvars, ['ansible_default_ipv4', 'address']) %}
+      host {{ postgres_db_name }} {{ postgres_db_username }} {{ host }}/32 md5
+      {% endfor %}
+    marker: >-
+      # {mark} ANSIBLE MANAGED: allow {{ postgres_db_clients | join(", ") }} to connect to {{ postgres_db_name }} as {{ postgres_db_username }}
+    owner: postgres
+    group: postgres
+    mode: "u=rw"
+
+- name: Configure PostgreSQL database
+  become: true
+  become_user: postgres
+  vars:
+    ansible_ssh_pipelining: true
```

Comment:
should not be declared here

Reply:
It is purposely enforced here to ensure the task would not fail even is user changes the ansible global  configuration to disable pipelining.
See https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_privilege_escalation.html#risks-of-becoming-an-unprivileged-user

Reply:
from the docs I am not understanding why pipelining should be enabled when becoming an unprivileged user

Reply:
to avoid permission issues when dealing with temp module files